include/PreferenceList.h

```diff
@@ -56,6 +59,17 @@ class PreferenceList {
     /// get pref list element at given index
     [[nodiscard]] PrefListElement at(SizeType index) const;
 
+    //set the ties at the given rank 
+    void set_ties(RankType rank, PreferenceList p); 
+
+    //get the tied elements at the given rank 
+    [[nodiscard]] std::vector<PrefListElement> get_ties(RankType rank) const; 
+
+    //find if the particular rank is tied 
+    bool isTied(RankType rank) const;
```

Comment:
Rename to is_tied. Leave a space between "//" and start of sentence, this applies for other places too.